### PR TITLE
QEA-1247: add refresh method and exception logging to page class

### PR DESCRIPTION
**Pull request by yet-another-lucas**
I needed a refresh method in the base page class because lots of mako tests will make api calls then refresh the page before continuing.

I added a unit test for this.

I also saw exceptions not being logged, so I added that.

### CODE REVIEW

File: lib/page.rb [hunk 1]
```diff
@@ -130,7 +134,8 @@ def click_button(button_name, button_index: nil)
       end
       begin
         button.click
-      rescue Exception
+      rescue Exception => exception
+        Log.debug("page.click_button exception was rescued: #{exception}")
```
Comment (clee) [inline lib/page.rb]: This seems a *little* redundant with the next line. Can we merge the two debug statements so we don't print two lines of debug output for this exception?

**Comment by sethuster** [comment]
Why not just call the Driver.refresh from the page object?

**Comment by yet-another-lucas** [comment]
@sethuster I should have put this bit from the jira story in the PR to explainL `while refactoring mako_nav I saw that it has a refresh method that does some boilerplate stuff in addition to the refresh`

File: lib/page.rb (hunk 1 above)
Comment (yet-another-lucas) [inline lib/page.rb]: This sounds good to me.

File: lib/page.rb [hunk 2]
```diff
@@ -22,7 +22,8 @@ def self.has_css?(css, options={})
       wait = Selenium::WebDriver::Wait.new(:timeout => 5)
       begin
         wait.until {Driver.driver.find_element(:css, css).enabled?}
-      rescue Exception
+      rescue Exception => exception
+        Log.debug("page.has_css? is false because this exception was rescued: #{exception}")
```
Comment (sethuster) [inline lib/page.rb]: Can we prefix these log messages with "[GRIDIUM::Page] " I like being able to see output like that in debug mode.

Comment (yet-another-lucas) [inline lib/page.rb]: Totally agree.

**Pull request merged**